### ISSUE TITLE
Glitched and flickering graphics

### ISSUE THREAD

**Issue by Missingno255** [MiSTer-devel/SMS_MiSTer#143]
Some games like, Addams Family, Terminator 2, and Taito Chase HQ, have flickering or glitched graphics.

The Addams Family has flickering graphics at the top of the screen about 15 or so pixels from the top
![20220822_203938-Addams Family, The (Europe)](https://user-images.githubusercontent.com/74580537/186066914-abe061f1-721b-40cf-81b3-09c0c6513dcf.png)
Correct
![20220822_205929-Addams Family, The (Europe)](https://user-images.githubusercontent.com/74580537/186066918-96fe2f67-49c7-49fc-b36b-f40ea6839acc.png)
incorrect

You can see this throughout the first level. It doesn't show very well in the images, but is very noticeable and reproducible. The main character also glitches out when his sprite is flipped horizontally and when walking. I couldn't capture it, but it's also fairly noticeable as artifacts are shown as he changes direction.
both problems are always very reproducible

Terminator 2
Arnold's face is....well not very flattering to say the least!

![20220822_204054-Terminator 2 - Judgment Day (Europe)](https://user-images.githubusercontent.com/74580537/186065916-f83efbd7-393b-43eb-9b58-ec86c6a977a9.png)
![20220822_204010-Terminator 2 - Judgment Day (Europe)](https://user-images.githubusercontent.com/74580537/186065923-e6d4e671-3f62-4cfa-9eb1-1aae66e4628e.png)
Always reproducible.

Taito Chase HQ
The graphics for the road and the HUD at the bottom of the screen when turning, flicker and glitch out. This was too hard for me to capture quickly enough, but maybe someone else can get a clear shot of this, as it's VERY noticeable and annoying and also causes performance issues, which you can hear the PSG getting screwed up.

**Comment by birdybro** [comment]
The flickering in Addams Family goes away if you set the TV System to PAL, that game was only released in PAL regions, this kind of thing happens on Genesis as well rarely. There is no region detection method for SMS games that is simple enough to be done efficiently, most software emulators get around this by using a large CRC32 database and do a quick match.

I easily reproduced the Terminator 2 error (another EU only game it seems).

Taito Chase HQ is the same as Addams Family, looks like you need to switch TV System to PAL in the Audio & Video options.

**Comment by Missingno255** [comment]
> The flickering in Addams Family goes away if you set the TV System to PAL, that game was only released in PAL regions, this kind of thing happens on Genesis as well rarely. There is no region detection method for SMS games that is simple enough to be done efficiently, most software emulators get around this by using a large CRC32 database and do a quick match.
> 
> I easily reproduced the Terminator 2 error (another EU only game it seems).
> 
> Taito Chase HQ is the same as Addams Family, looks like you need to switch TV System to PAL in the Audio & Video options.

Yeah, setting it to TV system to PAL solved it. I've been wanting to explore the Master System library, and I forgot that this was pretty much a PAL only console, like the Amiga.

As for Terminator 2, he's mostly fixed, but he still has a bar that goes through him. I don't know if there's another setting I need to set, or if if it's something else, like an emulation problem.

**Comment by birdybro** [comment]
The SMS wasn't pretty much PAL-only, just it had relatively more success in PAL regions (and the most success in Brazil, but that's not PAL, that's PAL-M which uses the NTSC 525-lines 30 FPS system with PAL color encoding).

There are however a good number of PAL-exclusives that is surprising in the library, yeah.

The Terminator 2 issue is likely an emulation problem. I wonder if it's related to this one --> https://github.com/MiSTer-devel/SMS_MiSTer/issues/113

**Comment by birdybro** [comment]
[SMS.zip](https://github.com/MiSTer-devel/SMS_MiSTer/files/9426661/SMS.zip)

Try this build for me please @Missingno255 

Thanks!

**Comment by Missingno255** [comment]
> [SMS.zip](https://github.com/MiSTer-devel/SMS_MiSTer/files/9426661/SMS.zip)
> 
> Try this build for me please @Missingno255 - Just a guess :)
> 
> Thanks!
Setting the Z80 mode to turbo, fixes the graphics.

**Comment by Missingno255** [comment]
> [SMS.zip](https://github.com/MiSTer-devel/SMS_MiSTer/files/9426661/SMS.zip)
> 
> Try this build for me please @Missingno255 - Just a guess :)
> 
> Thanks!

Setting the Z80 mode to turbo, fixes the graphics.

**Comment by birdybro** [comment]
The z80 set to turbo actually is not the normal speed. The z80 in the SMS runs at a slower speed than a normal z80. SMS z80 runs at 3.58MHz, the turbo mode brings it up to 5MHz which is similar to a mod you can do on original hardware by swapping oscillators.
